pandas/io/formats/format.py

```diff
@@ -565,315 +583,446 @@ def __init__(
     ):
         self.frame = frame
         self.show_index_names = index_names
+        self.sparsify = sparsify
+        self.float_format = float_format
+
+        # Ignoring error
+        # expression has type "Union[List[Callable[..., Any]],
+        # Tuple[Callable[..., Any], ...],
+        # Mapping[Union[str, int], Callable[..., Any]], None]",
+        # variable has type "Union[List[Callable[..., Any]],
+        # Tuple[Callable[..., Any], ...], Mapping[Union[str, int],
+        # Callable[..., Any]]]")
+        self.formatters = formatters  # type: ignore[assignment]
+        self.na_rep = na_rep
+        self.decimal = decimal
 
+        # Ignoring error
+        # expression has type "Union[str, int, Sequence[Union[str, int]],
+        # Mapping[Optional[Hashable], Union[str, int]], None]",
+        # variable has type "Mapping[Optional[Hashable], Union[str, int]]"
+        self.col_space = col_space  # type: ignore[assignment]
+
+        self.header = header
+        self.index = index
+        self.line_width = line_width
+        self.max_rows = max_rows
+        self.min_rows = min_rows
+        self.max_cols = max_cols
+        self.show_dimensions = show_dimensions
+        self.table_id = table_id
+        self.render_links = render_links
+        self.justify = justify
+        self.bold_rows = bold_rows
+        self.escape = escape
+
+        # Ignoring error:
+        # expression has type "Optional[Sequence[str]]", variable has type "Index"
+        self.columns = columns  # type: ignore[assignment]
+
+        self._truncate()
+        self.adj = get_adjustment()
+
+    @property
+    def sparsify(self):
+        return self._sparsify
+
+    @sparsify.setter
+    def sparsify(self, sparsify):
         if sparsify is None:
-            sparsify = get_option("display.multi_sparse")
+            self._sparsify = get_option("display.multi_sparse")
+        else:
+            self._sparsify = sparsify
 
-        self.sparsify = sparsify
+    @property
+    def formatters(self) -> FormattersType:
+        return self._formatters
 
-        self.float_format = float_format
+    @formatters.setter
+    def formatters(self, formatters: Optional[FormattersType]) -> None:
+        self._formatters: FormattersType
         if formatters is None:
-            self.formatters = {}
-        elif len(frame.columns) == len(formatters) or isinstance(formatters, dict):
-            self.formatters = formatters
+            self._formatters = {}
```

Comment:
i think if you type this it might be able to remove some of the mypy errors above (alt is to put in the class as typed)

Reply:
I replaced setters with initialization methods for each parameter. Now there is no problem with mypy.